test/Mono.Linker.Tests.Cases/DataFlow/LinkerAttributeRemoval.cs

```diff
@@ -41,4 +45,13 @@ public static void TestType ()
 			var typeKept = Type.GetType (reflectionTypeKeptString, false);
 		}
 	}
+
+	[KeptBaseType (typeof (System.Attribute))]
+	class TestDontRemoveAttribute:Attribute
```

Comment:
This is a good test, but we need another one - in the LinkerAttributeRemoval.xml try to add another attribute which will have effect on linker onto the attribute itself. This is tricky since we don't really have anything right now... maybe suppressmessage attribute would work.

Maybe add an issue to add this test as it needs a bit more work to figure out how to test it - basically we want to test that one can combine both normal attributes and internal attribute on the same element in the XML (And that it works).

Reply:
I opened #1323  to track that test